.CMake/compiler_opts.cmake

```diff
@@ -117,6 +118,9 @@ elseif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
     add_compile_options(-Wformat=2)
     add_compile_options(-Wfloat-equal)
     add_compile_options(-Wwrite-strings)
+    if (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
Is this guard necessary? Why then is it not active for Clang above?

Reply:
Yes it's necessary (I get a build failure if it's not there). But it is not necessary for clang above -- and I can't explain why. All I know at this point is that this configuration seems to work.